qiskit/circuit/library/generalized_gates/linear_function.py

```diff
@@ -171,6 +171,18 @@ def permutation_pattern(self):
         locs = np.where(linear == 1)
         return locs[1]
 
+    def permute(self, perm):
```

Comment:
I think that it's worth to add a test for this.

Reply:
Good point, adding this test.